frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -678,6 +678,13 @@ pub mod pallet {
 			let _: UpperOf<CompactAccuracyOf<T>> = maximum_chain_accuracy
 				.iter()
 				.fold(Zero::zero(), |acc, x| acc.checked_add(x).unwrap());
+
+			// We only accept data provider who's maximum votes per voter matches our
+			// `T::CompactSolution`'s `LIMIT`.
+			assert_eq!(
```

Comment:
note that the main reason why I need to add `MAXIMUM_VOTES_PER_VOTER` const is so that I can make this assertion -- otherwise this might be a foot gun.